src/ucp/core/ucp_device.c

```diff
@@ -258,25 +194,13 @@ ucp_device_mem_list_params_check(ucp_context_h context,
             return status;
         }
 
-        status = ucp_check_memh_elem(element, i, local_sys_dev,
-                                     local_md_map, mem_type);
+        status = ucp_check_memh_elem(element, i, local_sys_dev, local_md_map,
+                                     &first_memh);
         if (status != UCS_OK) {
             return status;
         }
     }
 
-    if (*local_sys_dev == UCS_SYS_DEVICE_ID_UNKNOWN) {
-        *mem_type = UCS_MEMORY_TYPE_CUDA;
-        status    = ucp_device_detect_local_sys_dev(context, *mem_type,
-                                                    local_sys_dev);
-        if (status != UCS_OK) {
-            return status;
-        }
-
-        *local_md_map = ucp_device_detect_local_md_map(context,
-                                                       *local_sys_dev);
-    }
```

Comment:
IMO we cannot just remove this because mem_list itself may be allocated in wrong place